lib/content_location_filter.js

```diff
@@ -0,0 +1,42 @@
+"use strict";
+
+const HyperSwitch = require('hyperswitch');
+const Template = HyperSwitch.Template;
+const URI = HyperSwitch.URI;
+const mwUtil = require('./mwUtil');
+
+let basePathTemplate;
+
+module.exports = (hyper, req, next, options) => {
+    basePathTemplate = basePathTemplate || new Template({
+        uri: options.templates.base_uri_template
+    });
+    if (req.method !== 'get') {
+        return next(hyper, req);
+    } else {
+        const baseUri = basePathTemplate.expand({
+            request: {
+                params: {
+                    // This is a bit of a hack, but we know the domain
+                    // will be the first in the path, so use that info
+                    domain: req.uri.path[0]
```

Comment:
This makes me wonder if the /v1/ route filter would be better overall. In that case, we should be able to use `req.params.domain`. Another benefit of a route filter on /v1/ would be that it would apply to internal sub-requests to public end points as well.